.github/workflows/builder.yaml

```diff
@@ -0,0 +1,57 @@
+name: CI Build & Push Builder Image
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths:
+    - Dockerfile.builder
+    - .bazelversion
+
+jobs:
+  build-and-push-builder:
+    timeout-minutes: 360
+    name: Build and push multi-arch builder images
+    runs-on: ubuntu-latest-64-cores-256gb
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@8c0edbc76e98fa90f69d9a2c020dcb50019dc325
+        with:
+          version: v0.9.1
+      - name: Cache Docker layers
+        uses: actions/cache@58c146cc91c5b9e778e71775dfe9bf1442ad9a12
+        with:
+          path: /tmp/buildx-cache
+          key: docker-cache-${{ github.head_ref }}
+          restore-keys: docker-cache-master
+      - name: Login to quay.io
+        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_ENVOY_USERNAME_DEV }}
+          password: ${{ secrets.QUAY_ENVOY_PASSWORD_DEV }}
+      - name: Checkout PR
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Prep for build
+        run: |
+          echo "${{ github.event.pull_request.head.sha }}" >SOURCE_VERSION
+          echo "ENVOY_VERSION=$(cat ENVOY_VERSION)" >> $GITHUB_ENV
+          echo "BAZEL_VERSION=$(cat .bazelversion)" >> $GITHUB_ENV
+      - name: PR Multi-arch build & push of cilium-envoy-builder
+        uses: docker/build-push-action@37abcedcc1da61a57767b7588cb9d03eb57e28b3
+        id: docker_build_builder_ci
+        with:
+          context: .
+          file: ./Dockerfile.builder
+          platforms: linux/amd64,linux/arm64
+          build-args: |
+            BAZEL_BUILD_OPTS=--remote_upload_local_results=false
+          cache-from: type=local,src=/tmp/buildx-cache
+          cache-to: type=local,dest=/tmp/buildx-cache,mode=max
```

Comment:
Docker caching may interfere with actually pulling the latest packages, so maybe not ask for it?